core/shared/src/main/scala/com/lynbrookrobotics/potassium/streams/ExpectedPeriodicity.scala

```diff
@@ -0,0 +1,9 @@
+package com.lynbrookrobotics.potassium.streams
+
+import squants.time.Time
+
+sealed trait ExpectedPeriodicity
+
+case class Periodic(period: Time) extends ExpectedPeriodicity
```

Comment:
Because `Stream`s can be transformed in ways that do not maintain their periodicity (because of the push flow), we need to keep track of if a `Stream` is periodic and if so what its period is.